commands/delegate_submission.go

```diff
@@ -0,0 +1,49 @@
+package commands
+
+import (
+	commandspb "code.vegaprotocol.io/go-wallet/internal/proto/commands/v1"
+)
+
+func CheckDelegateSubmission(cmd *commandspb.DelegateSubmission) error {
+	return checkDelegateSubmission(cmd).ErrorOrNil()
+}
+
+func checkDelegateSubmission(cmd *commandspb.DelegateSubmission) Errors {
+	errs := NewErrors()
+
+	if cmd == nil {
+		return errs.FinalAddForProperty("delegate_submission", ErrIsRequired)
+	}
+
+	if cmd.Amount <= 0 {
+		errs.AddForProperty("delegate_submission.amount", ErrIsRequired)
+	}
+
+	if len(cmd.NodeId) <= 0 {
+		errs.AddForProperty("delegate_submission.node_id", ErrIsRequired)
+	}
+
+	return errs
+}
+
+func CheckUndelegateAtEpochEndSubmission(cmd *commandspb.UndelegateAtEpochEndSubmission) error {
+	return checkUndelegateAtEpochEndSubmission(cmd).ErrorOrNil()
+}
+
+func checkUndelegateAtEpochEndSubmission(cmd *commandspb.UndelegateAtEpochEndSubmission) Errors {
+	errs := NewErrors()
+
+	if cmd == nil {
+		return errs.FinalAddForProperty("undelegateAtEpochEnd_submission", ErrIsRequired)
+	}
+
+	if cmd.Amount <= 0 {
```

Comment:
If the user specifies a negative amount, the error message is going to be `amount is required`, which doesn't aligned. You can use `ErrMustBePositive` instead.

Reply:
Does positive imply non-zero? Because in this case I'd expect to be a non zero positive value.

Reply:
Same as above, it's an unsigned type so no negatives.